### DISCUSSION TITLE
Connection is busy with results for another command (0) (SQLExecDirectW)

### DISCUSSION THREAD

**Discussion by soujanyat** [mkleehammer/pyodbc#1334] [Q&A]
Please first make sure you have looked at:

* Documentation: https://github.com/mkleehammer/pyodbc/wiki
* Other issues

### Environment
- Python: 3.10.9
- pyodbc: 4.0.35
- OS: Ubuntu 20.04
- DB: MSSQL
- driver: ODBC Driver 17 for SQL Server

### Issue
sqlalchemy.exc.DBAPIError: (pyodbc.Error) ('HY000', '[HY000] [Microsoft][ODBC Driver 17 for SQL Server]Connection is busy with results for another command (0) (SQLExecDirectW)'

SQLAlchemy - 1.4.52
zope.sqlalchemy - 3.1
z3c.sqlalchemy = 2.1.1
No of Threads = 4

We are working on SQLAlchemy version up gradation project from version 1.3.x to 1.4.52 on ZOPE Framework. Post upgrade started getting Connection busy error and unable to access the application. We need your help to understand the issue and solution.

Thank You,

**Comment by v-chojas** [comment]
Try adding MARS_Connection=Yes to your connection string.

**Comment by soujanyat** [comment]
> Try adding MARS_Connection=Yes to your connection string.

Thank you @v-chojas 
yes, we have tested with MARS_Connection=Yes connection string. Connection busy error is resolved however faced application slowness more than thrice the execution time of each APIs.

Is there a way to handle it without MARS_Connection or handle slowness

**Comment by v-chojas** [comment]
It sounds like your application is attempting to use multiple active resultsets, so you'll need to figure out how to change it to not do that. Unless MARS is enabled, the driver can only process one query at a time on a connection.
